### PR TITLE
Explain how to run the unit tests

### PR DESCRIPTION
**Pull request by Johennes**
This ports the documentation improvements from #1080 over to @pixlwave's branch for #1023.

### CODE REVIEW

File: bindings/apple/README.md [hunk 1]
```diff
@@ -38,15 +38,19 @@ The `build_crypto_xcframework.sh` script will go through all the steps required
...
-It's pre-configured to link to the generated .xcframework and .swift files so successfully running the script first is necessary for it to compile.
+It's pre-configured to link to the generated .xcframework and .swift files so successfully running one of the scripts first is necessary for it to compile.
 
-It makes the compiled code available to swift by importing the C header through its bridging header.
+If you're using `debug_build_xcframework.sh`, you can use `xcodebuild` to execute the unit tests in an iOS simulator, e.g.
 
-Once all the generated components are available running it should be as easy as choosing a platform and clicking run.
+```
+xcodebuild test -scheme MatrixRustSDK -sdk iphonesimulator -destination 'platform=iOS Simulator,name=iPhone 13'
+```
```
Comment (pixlwave) [inline bindings/apple/README.md:51]: So this won't be an option for running the tests anymore. The package is only configured to run on the current platform. So its `debug_build_library.sh` and `swift test` or nothing.

Comment (Johennes) [reply bindings/apple/README.md:51]: Ah so you're planning to entirely remove the `debug_build_xcframework.sh` script with your PR?

Comment (pixlwave) [reply bindings/apple/README.md:51]: Not quite. The `debug_build_xcframework.sh` will remain as we still need that for consuming debug builds on EX. The `debug_build_library.sh` script is purely for the tests Swift package.

Comment (pixlwave) [reply bindings/apple/README.md:51]: Although, in the future if we hook the static lib build into Xcode's build process, we might be able to get rid of debug XCFrameworks and just use them for distribution.

Comment (Johennes) [reply bindings/apple/README.md]: Sorry, I was lacking coffee this morning. 🤦‍♂️ Understood now that you were referring to the removal the iOS platform from `Package.swift`.

I've removed mentions of `debug_build_xcframework.sh` and the command to run the tests in the simulator now.

File: bindings/apple/README.md [hunk 2]
```diff
@@ -38,15 +38,11 @@ The `build_crypto_xcframework.sh` script will go through all the steps required
...
 
-The Xcode project is meant to provide a simple example on how to integrate everything together but also a place to run unit and integration tests from.
+`Package.swift` is meant to provide a simple example on how to integrate everything together but also a place to run unit and integration tests from.
 
-It's pre-configured to link to the generated .xcframework and .swift files so successfully running the script first is necessary for it to compile.
-
-It makes the compiled code available to swift by importing the C header through its bridging header.
-
-Once all the generated components are available running it should be as easy as choosing a platform and clicking run.
+It's pre-configured to link to the generated .xcframework and .swift files so successfully running the `debug_build_library.sh` script first is necessary for it to compile. Afterwards you can execute the tests with `swift test`. Note that for the moment this only works on macOS but we're planning to add Linux support in the future.
```
Comment (pixlwave) [inline bindings/apple/README.md:45]:
```suggestion
It's pre-configured to link to the generated static lib and .swift files so successfully running the `debug_build_library.sh` script first is necessary for it to compile. Afterwards you can execute the tests with `swift test`. Note that for the moment this only works on macOS but we're planning to add Linux support in the future.
```

**Review by pixlwave — APPROVED** [review]
LGTM 👍

File: bindings/apple/README.md (hunk 2 above)
Comment (Johennes) [reply bindings/apple/README.md:45]: Oh, good catch!

**Pull request merged**